src/io/xml.c

```diff
@@ -112,14 +116,33 @@ const char *get_status_string(struct criterion_test_stats *ts,
     return status;
 }
 
+/*
+ * floats are printed locale dependent, but the xml-specification
+ * requires a dot as the decimal separator.
+ * Therefore we set the locale temporarily to print dots.
+ */
+static int fprintf_locale(FILE *__restrict __stream,
```

Comment:
use `restrict` instead of `__restrict`, and don't use leading underscore in identifiers